tools/interop_matrix/ruby_v101_build_interop_patch.patch

```diff
@@ -0,0 +1,11 @@
+--- build_interop	2017-11-15 00:27:57.478490290 +0000
```

Comment:
I think the patches belong to some subdirectory (e.g. `tools/interop_matrix/patches`  which would preferrably have a README.md) rather than directly under tools/interop_matrix.

Reply:
Maybe under tools/interop_matrix/patches/ruby_v1.0.1, with a README about why we need it.

Reply:
@adelez went with your directory suggestion.

Modified the script to follow the `patches/<lang>_<release>/git_repo.patch` convention in general as a way to find patches.

Added README under `patches`